src/app/random-name/random-name.component.ts

```diff
@@ -1,24 +1,35 @@
 import { Component, OnInit } from '@angular/core';
 import { RandomNameService } from './random-name.service';
+import { Observable } from 'rxjs';
+import { map, catchError } from 'rxjs/operators';
+import { Name, NamesData } from './names';
 
 @Component({
   selector: 'app-random-name',
   templateUrl: './random-name.component.html',
   styleUrls: ['./random-name.component.css']
 })
 export class RandomNameComponent implements OnInit {
-  public generatedName: string;
+  public errorMessage: string = '';
+  public $generatedName: Observable<Name>;
 
-  constructor(private randomName: RandomNameService) { }
+  constructor(private randomNameService: RandomNameService) {}
 
-  public showGeneratedName() {
-    this.randomName.getNames()
-      .subscribe((data: any[]) => {
-        this.generatedName = data[Math.floor(Math.random() * data.length)];
-      });
+  ngOnInit() {
+    this.load();
   }
 
-  ngOnInit() {
-    this.showGeneratedName();
+  load() {
+    this.$generatedName = this.randomNameService
```

Comment:
Metoda load nie jest tu potrzebna. Jak stosujesz taką logikę na onInit to juz ją zostaw w onInit. Tą metodę możesz użyć tylko raz i tylko w onInit.

Reply:
Przerzuciłem generowanie random name do serwisu, a w componencie tylko przypisuję stream